config/src/main/java/com/alibaba/nacos/config/server/controller/ConfigController.java

```diff
@@ -382,4 +388,229 @@ public GroupkeyListenserStatus getListeners(HttpServletRequest request, HttpServ
+                                               @RequestParam(value = "ids", required = false)List<Long> ids) {
+        String idsStr = idList2String(ids);
+        List<ConfigInfo> dataList = persistService.findAllConfigInfo4eExport(group, tenant, appName, idsStr);
+        List<ZipUtils.ZipItem> zipItemList = new ArrayList<>();
+        StringBuilder metaData = null;
+        for(ConfigInfo ci : dataList){
+            if(StringUtils.isNotBlank(ci.getAppName())){
+                // Handle appName
+                if(metaData == null){
+                    metaData = new StringBuilder();
+                }
+                String metaDataId = ci.getDataId();
+                if(metaDataId.contains(".")){
+                    metaDataId = metaDataId.substring(0,metaDataId.lastIndexOf("."))
+                        + "~" + metaDataId.substring(metaDataId.lastIndexOf(".") + 1);
+                }
+                metaData.append(ci.getGroup()).append(".").append(metaDataId).append(".app=")
+                    // Fixed use of "\r\n" here
+                    .append(ci.getAppName()).append("\r\n");
+            }
+            String itemName = ci.getGroup() + "/" + ci.getDataId() ;
+            zipItemList.add(new ZipUtils.ZipItem(itemName, ci.getContent()));
+        }
+        if(metaData != null){
+            zipItemList.add(new ZipUtils.ZipItem(".meta.yml", metaData.toString()));
+        }
+
+        HttpHeaders headers = new HttpHeaders();
+        String fileName="nacos_config_export_" + DateFormatUtils.format(new Date(), "yyyy-MM-dd HH:mm:ss") + ".zip";
+        headers.add("Content-Disposition", "attachment;filename="+fileName);
+        return new ResponseEntity<byte[]>(ZipUtils.zip(zipItemList), headers, HttpStatus.OK);
+    }
+
+    @RequestMapping(params = "import=true", method = RequestMethod.POST)
+    @ResponseBody
+    public RestResult<Map<String, Object>> importAndPublishConfig(HttpServletRequest request, HttpServletResponse response,
+                                                                  @RequestParam(value = "src_user", required = false) String srcUser,
+                                                                  @RequestParam(value = "namespace", required = false) String namespace,
+                                                                  @RequestParam(value = "policy", defaultValue = "ABORT")
+                                                                          SameConfigPolicy policy,
+                                                                  MultipartFile file) throws NacosException {
+        RestResult<Map<String, Object>> rr = new RestResult<>();
+        Map<String, Object> failedData = new HashMap<>(4);
+        rr.setData(failedData);
+
+        if(StringUtils.isNotBlank(namespace)){
+            if(persistService.tenantInfoCountByTenantId(namespace) <= 0){
+                rr.setCode(5001);
```

Comment:
Here are the following two questions:
1. Please use a meaningful name for the code field
``` java
XXX_XXX = 5001
```
2. This repeated operation can be extracted as a method of reuse
```suggestion
                // rr.setCode(5001);
// failedData.put("succCount", 0);
// rr.setMessage("namespace does not exist");

buildResult(Integer code,Map<String,Object> data,String message);
```